synthesizer/src/coinbase_puzzle/mod.rs

```diff
@@ -209,6 +211,17 @@ impl<N: Network> CoinbasePuzzle<N> {
         };
 
         // Construct the provers polynomial.
+        #[cfg(feature = "serial")]
+        let accumulated_prover_polynomial = cfg_into_iter!(prover_polynomials).zip_eq(challenges).fold(
```

Comment:
This was always being run in parallel so it would break if you tried to use this with default features so now we can handle both serial and parallel executions.

Reply:
Nice

Reply:
Perhaps we should write some tests to verify this? I can help with this.